modules/tools/decode_video/video2jpg.cc

```diff
@@ -0,0 +1,117 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+
+#include "gflags/gflags.h"
+
+#include "cyber/common/log.h"
+#include "modules/tools/decode_video/h265_decoder.h"
+
+DECLARE_string(input_video);
+DECLARE_string(output_dir);
+DEFINE_string(input_video, "", "The input video file");
+DEFINE_string(output_dir, "", "The directory to output decoded pictures.");
+
+std::vector<uint8_t> read_video_file(const std::string file_name) {
+  std::ifstream in(file_name, std::ios::binary);
+  std::vector<uint8_t> file_buffer((std::istreambuf_iterator<char>(in)),
+                                   std::istreambuf_iterator<char>());
+  return file_buffer;
+}
+
+void write_jpg_file(const std::vector<uint8_t>& jpeg_buffer,
+                    const std::string output_dir, const int frame_num) {
+  const int JPEG_FILE_LEN = 256;
+  const int FILE_SUFFIX_LEN = 16;
+  const std::string output_file_pattern = output_dir + "/%05d.jpg";
+  char jpeg_file_path[JPEG_FILE_LEN];
+  snprintf(jpeg_file_path, output_file_pattern.length() + FILE_SUFFIX_LEN,
+           output_file_pattern.c_str(), frame_num);
+  std::ofstream out(jpeg_file_path, std::ios::binary);
+  for (auto&& current : jpeg_buffer) out << static_cast<char>(current);
+}
+
+int stream_decoder_process_frame(const std::vector<uint8_t>& indata,
```

Comment:
Put non-class-member functions into anonymous namespace.

Reply:
now they are all moved to class functions